georest/__init__.py

```diff
@@ -286,7 +286,7 @@ def add_file_to_mosaic(config, fname_in, filesystem='posix'):
     This function wraps some boilerplate around adding a granule to a layer.
 
     """
-    fname = utils.convert_file_path(config, fname_in)
+    fname = utils.convert_file_path(config, fname_in, keep_subpath=config["keep_subpath"])
```

Comment:
Could also be
```suggestion
    fname = utils.convert_file_path(config, fname_in, keep_subpath=config.get("keep_subpath", False))
```
so forgetting the extra option from the config doesn't cause a crash.